src/main/kotlin/no/skatteetaten/aurora/boober/feature/FluentbitSidecarFeature.kt

```diff
@@ -0,0 +1,354 @@
+            listOf(
+                AuroraConfigFieldHandler(
+                    "logging/custom/$key/index",
+                    defaultValue = header.loggingIndex
+                ),
+                AuroraConfigFieldHandler(
+                    name = "logging/custom/$key/pattern"
+                    // validator =
+                    // TODO validate that this is requered
+                ),
+                AuroraConfigFieldHandler(
+                    name = "logging/custom/$key/sourcetype"
+                    // validator =
+                    // TODO validate that this is requered
+                )
+            )
+        }
+
+        val loggerHanlders = knownLogs.map { log ->
+            AuroraConfigFieldHandler("logging/loggers/$log")
+        }
+
+        return listOf(
+            AuroraConfigFieldHandler("logging/index")
+        ).addIfNotNull(customHandlers)
+            .addIfNotNull(loggerHanlders)
+            .toSet()
+    }
+
+    override fun generate(adc: AuroraDeploymentSpec, cmd: AuroraContextCommand): Set<AuroraResource> {
+        val index = adc.loggingIndex ?: return emptySet()
+        val loggerIndexes = getLoggingIndexes(adc, cmd, index)
+
+        val fluentParserMap = newConfigMap {
+            metadata {
+                name = adc.fluentParserName
+                namespace = adc.namespace
+            }
+            data = mapOf(parsersFileName to generateParserConf())
+        }
+
+        val fluentConfigMap = newConfigMap {
+            metadata {
+                name = adc.fluentConfigName
+                namespace = adc.namespace
+            }
+            data = mapOf("fluent-bit.conf" to generateFluentBitConfig(loggerIndexes, adc.name, adc.cluster))
+        }
+
+        val hecSecret = newSecret {
+            metadata {
+                name = adc.hecSecretName
+                namespace = adc.namespace
+            }
+            data = mapOf(hecTokenKey to Base64.encodeBase64String(hecToken.toByteArray()),
+                    splunkHostKey to Base64.encodeBase64String(splunkUrl.toByteArray()),
+                    splunkPortKey to Base64.encodeBase64String(splunkPort.toByteArray()))
+        }
+        return setOf(generateResource(fluentParserMap), generateResource(fluentConfigMap), generateResource(hecSecret))
+    }
+
+    override fun modify(adc: AuroraDeploymentSpec, resources: Set<AuroraResource>, cmd: AuroraContextCommand) {
+        if (adc.loggingIndex == null) return
+
+        val configVolume = newVolume {
+            name = adc.fluentConfigName
+            configMap {
+                name = adc.fluentConfigName
+            }
+        }
+        val parserVolume = newVolume {
+            name = adc.fluentParserName
+            configMap {
+                name = adc.fluentParserName
+            }
+        }
+
+        val container = createFluentbitContainer(adc)
+        resources.forEach {
+            if (it.resource.kind == "DeploymentConfig") {
```

Comment:
Akkurat nå er denne featuren enablet for jobs/cronjobs også. Så enten må man skru den av for disse eller så må man modifisere jobs/cronjobs også.

Reply:
Godt poeng! Skal se hvordan jeg kan filtrere bort jobs/cronjobs.

Reply:
Men hvis TemplateType er job og cronjobs vil den vel ikke være enablet? Har lagt til test om existerende logging config og om det er en jobb, men sistnevnte er vel egentlig overflødig.

Reply:
Her har jeg lest koden feil. Du har rett.